kratos/mpi/tests/cpp_tests/utilities/test_pointer_communicator.cpp

```diff
@@ -85,6 +86,71 @@ KRATOS_TEST_CASE_IN_SUITE(PointerCommunicator, KratosMPICoreFastSuite)
     }
 }
 
+KRATOS_TEST_CASE_IN_SUITE(PointerCommunicatorPartialPartitions, KratosMPICoreFastSuite)
+{
+    DataCommunicator& r_default_comm = ParallelEnvironment::GetDefaultDataCommunicator();
+    Model current_model;
+    auto& mp = current_model.CreateModelPart("mp");
+    mp.AddNodalSolutionStepVariable(PARTITION_INDEX);
+    mp.AddNodalSolutionStepVariable(TEMPERATURE);
+
+    const int world_size = r_default_comm.Size();
+    const int current_rank = r_default_comm.Rank();
+
+    auto pnode = mp.CreateNewNode(current_rank+1, current_rank,current_rank,current_rank); //the node is equal to the current rank;
+    pnode->FastGetSolutionStepValue(PARTITION_INDEX) = current_rank;
+    pnode->SetValue(TEMPERATURE, current_rank );
+
+    //we will gather on every node the global pointers of the nodes with index from
+    //current_rank(+1) to world_size
+    std::vector<int> indices, ranks;
+    indices.reserve(world_size - current_rank - 1);
+    for(int i = current_rank + 1; i < world_size; ++i) {
+        indices.push_back(i);
+    }
+    ranks.reserve(world_size - 1);
+    for(int i = 0; i < world_size - 1; ++i) {
+        ranks.push_back(i);
+    }
+
+    if (current_rank < world_size - 1) {
+        auto& r_partial_data_comm = r_default_comm.GetSubDataCommunicator(ranks, "SubDataComm");
```

Comment:
can you please a tests also for the error conditions
and one for calling this 2 times, i.e. creating a sub-sub datacomm

Reply:
Okay

Reply:
AT least this is done